Fix/10376/correct display names

caybro opened the pull request:
### What does the PR do

Fixes several places where the display name variations or changes thereto (nickname, ens name, etc) are not handled correctly.

(see the individual commits for explanation)

Fixes: #10376

### Affected areas

ActivityCenter, ProfileDialog and places involving StatusMemberListItem

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

[Záznam obrazovky z 2023-04-19 20-08-02.webm](https://user-images.githubusercontent.com/5377645/233162958-8f607355-3b80-44ef-ab71-822019dab669.webm)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8c0eb1c7 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-10377/1/) | 2023-04-19 16:58:55 | ~6 min | `tests/imports` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-10377/1//consoleText) |

jrainville commented on ui/imports/shared/views/ExistingContacts.qml:
```diff
@@ -83,21 +83,21 @@ Item {
         delegate: StatusMemberListItem {
             width: contactListView.availableWidth
-
+            readonly property bool ensVerified: Utils.isEnsVerified(model.pubKey)
```
It would be even better if `ensVerified` was part of the model. It should be easy, since `createItemFromPublicKey` uses `getContactDetails` to populate the Item.
Also, if you do that, can you clean it up a little, we do both of those:
```nim
let contact =  self.controller.getContact(publicKey)
let contactDetails = self.controller.getContactDetails(contact.id)
```
which is useless, they basically return the same thing (only `contactDetails`  is useful, since`contactDetails.details` is the ContactDto)

caybro replied:
Yup, will do

jrainville reviewed (commented):
Code looks good. Improving the model just a little will help a lot, since we won't need to call `Utils.isEnsVerified`

MishkaRogachev commented on ui/app/mainui/Popups.qml:
```diff
@@ -362,6 +362,7 @@ QtObject {
                 onEditDone: {
                     if (nickname !== newNickname) {
                         rootStore.contactStore.changeContactNickname(publicKey, newNickname)
+                        Global.contactRenamed(publicKey)
```
There are signals in contacts module `contactNicknameChanged` and `contactUpdated` which can also trigger a change of localNickname and other names. Probably it would better to handle changes from status-go and ui in one place

caybro replied:
The problem is that not all people are contacts (e.g. added contacts), this I haven't figured out how to do.

MishkaRogachev reviewed (approved):
LGTM

caybro commented:
@jrainville @MishkaRogachev the last commit grew up a bit :) but I addressed your comment and:

- added `isEnsVerified` to the respective models
- fixed returning "ensName" when the user is not ensVerified
- fixed propagating contact (nick)names in the contacts module
- fix sorting in the user/member QML list view

jrainville reviewed (approved):
Way cleaner :clap:

caybro commented on src/app/modules/shared_models/user_item.nim:
```diff
@@ -170,7 +182,7 @@ proc `colorId=`*(self: UserItem, value: int) {.inline.} =
 
 proc colorHash*(self: UserItem): string {.inline.} =
-  self.colorHash
+  if not self.isEnsVerified: self.colorHash else: ""
```
Also, noticed the channel switching gets even a bit more faster now: probably due to the fewer side calls to `Utils.foo`

jrainville replied:
> Also, noticed the channel switching gets even a bit more faster now: probably due to the fewer side calls to `Utils.foo`

Yes, that's why I suggested putting stuff in the model. Each call we reduce from QML to Nim is a win for performance :trophy:

caybro replied:
About the `colorHash`, we do the same here as well:

```js
function getColorHashAsJson(publicKey, skipEnsVerification=false) {
        if (publicKey === "" || !isChatKey(publicKey))
            return
        if (skipEnsVerification) // we know already the user is ENS verified -> no color ring
            return
        if (isEnsVerified(publicKey)) // ENS verified -> no color ring
            return
        let jsonObj = globalUtilsInst.getColorHashAsJson(publicKey)
        return JSON.parse(jsonObj)
    }
```
